torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -3059,12 +3040,12 @@ def _post_backward_hook(
         with torch.autograd.profiler.record_function(
             "FullyShardedDataParallel._post_backward_hook"
         ):
-            # First hook callback will see PRE state. If we have multiple params,
-            # then subsequent hook callbacks will see POST state.
-            self._assert_state(
-                [TrainingState_.BACKWARD_PRE, TrainingState_.BACKWARD_POST]
+            self._assert_state([TrainingState.FORWARD_BACKWARD])
+            self.training_state = TrainingState.FORWARD_BACKWARD
```

Comment:
nit: why it is assigned the same state after it checked the state == TrainingState.FORWARD_BACKWARD

Reply:
Good point. I refactored too fast and overlooked this redundancy :)